server/utils/utils.test.ts

```diff
@@ -48,594 +49,638 @@ import {
-          agencyId: 'MDI',
-          assignedLivingUnitDescription: 'NMI-RECP',
-          nonAssociationName: 'John Doe',
-          offenderNo: 'ABC123',
-          reasonDescription: 'Victim',
+          id: 'id1',
+          desc: 'desc1',
+          random: 'random1',
         },
         {
-          agencyId: 'MDI',
-          assignedLivingUnitDescription: 'NMI-RECP',
-          nonAssociationName: 'Guy Incognito',
-          offenderNo: 'DEF321',
-          reasonDescription: 'Rival Gang',
+          id: 'id2',
+          desc: 'desc2',
+          random: 'random2',
         },
-      ])
-      expect(res).toEqual([
-        [
-          { html: '<a class="govuk-link govuk-link--no-visited-state" href="/prisoner/ABC123">John Doe</a>' },
-          { text: 'ABC123' },
-          { text: 'NMI-RECP' },
-          { text: 'Victim' },
-        ],
-        [
-          { html: '<a class="govuk-link govuk-link--no-visited-state" href="/prisoner/DEF321">Guy Incognito</a>' },
-          { text: 'DEF321' },
-          { text: 'NMI-RECP' },
-          { text: 'Rival Gang' },
-        ],
+      ]
+      const selectOptions = objectToSelectOptions(data, 'id', 'desc')
+
+      expect(selectOptions).toEqual([
+        { value: 'id1', text: 'desc1' },
+        { value: 'id2', text: 'desc2' },
       ])
     })
   })
-})
-
-describe('refDataToSelectOptions', () => {
-  it('should map ref data objects to select options', () => {
-    const refData: ReferenceCode[] = [
-      {
-        code: 'code1',
-        description: 'description1',
-        activeFlag: 'Y',
-      },
-      {
-        code: 'code2',
-        description: 'description2',
-        activeFlag: 'Y',
-      },
-    ]
-    const selectOptions = refDataToSelectOptions(refData)
 
-    expect(selectOptions).toEqual([
-      { value: 'code1', text: 'description1' },
-      { value: 'code2', text: 'description2' },
-    ])
+  describe('format community manager', () => {
+    it.each([
+      ['Staff recorded', { name: { forename: 'JOHN', surname: 'SMITH' } } as CommunityManager, 'John Smith'],
+      [
+        'Staff unallocated',
+        { team: { description: 'Probation Team' }, unallocated: true },
+        'Probation Team (COM not yet allocated)',
+      ],
+      ['404', null, 'Not assigned'],
+      ['Error', undefined, 'We cannot show these details right now'],
+    ])('%s: formatCommunityManager(%s, %s, %s)', (_: string, communityManager: CommunityManager, expected: string) => {
+      expect(formatCommunityManager(communityManager)).toEqual(expected)
+    })
   })
-})
 
-describe('objectToSelectOptions', () => {
-  it('should map objects to select options', () => {
-    const data: object[] = [
-      {
-        id: 'id1',
-        desc: 'desc1',
-        random: 'random1',
-      },
-      {
-        id: 'id2',
-        desc: 'desc2',
-        random: 'random2',
-      },
-    ]
-    const selectOptions = objectToSelectOptions(data, 'id', 'desc')
+  describe('groupBy', () => {
```

Comment:
Here we go - the actual change.

There was no test for the groupBy function.  There are a couple of edge cases that have therefore fallen through the cracks which I think need attention so I'll try to investigate as part of this ticket.